server/modules/surveyRdb/schemaRdb/dataView.js

```diff
@@ -23,91 +23,93 @@ export const columns = {
   keys: '_keys',
 }
 
-export const getColUuid = nodeDef => `${NodeDef.getName(nodeDef)}_${DataTable.colNameUuuid}`
+export const getColUuid = (nodeDef) => `${NodeDef.getName(nodeDef)}_${DataTable.colNameUuuid}`
 
-export const getSelectFields = (survey, nodeDef, resultStepViews) => {
+const _getCols = (survey, nodeDef, isSelf) => {
+  const fields = R.pipe(
+    R.map(DataCol.getNames),
+    R.flatten,
+    R.map((name) => `${isSelf ? alias : aliasParent}.${name}`)
+  )(DataTable.getNodeDefColumns(survey, nodeDef))
+
+  // If is not root, prepend parent uuid
+  if (!NodeDef.isRoot(nodeDef)) {
+    fields.unshift(`${aliasParent}.${getColUuid(Survey.getNodeDefParent(nodeDef)(survey))}`)
+  }
+
+  // If nodeDef isSelf (starting nodeDef) prepend col uuid
+  if (isSelf) {
+    fields.unshift(`${alias}.${DataTable.colNameUuuid} as ${getColUuid(nodeDef)}`)
+  }
+
+  return fields
+}
+
+export const getSelectFields = (survey, nodeDef, resultStepViews = [], excludeFieldKeys = false) => {
   if (NodeDef.isVirtual(nodeDef)) {
     return ['*']
   }
 
   const fields = []
 
-  Survey.visitAncestorsAndSelf(nodeDef, nodeDefCurrent => {
+  Survey.visitAncestorsAndSelf(nodeDef, (nodeDefCurrent) => {
     // Do not include node defs of calculation steps
-    const cols = getCols(survey, nodeDefCurrent, NodeDef.isEqual(nodeDefCurrent)(nodeDef))
+    const cols = _getCols(survey, nodeDefCurrent, NodeDef.isEqual(nodeDefCurrent)(nodeDef))
     fields.unshift(...cols)
   })(survey)
 
-  const nodeDefsCalculationColNames = R.pipe(
-    R.map(ResultStepView.getNodeDefColumns),
-    R.flatten,
-    R.map(DataCol.getNames),
-    R.flatten,
-  )(resultStepViews)
+  if (!R.isEmpty(resultStepViews)) {
```

Comment:
you added this just because of convenience of the select for the rData. you need to use instead some method that gives you only the nodeDef columns

Reply:
right